File: Projects/P2-Final/employee_app/config/app_config.py
import os
from pathlib import Path

from dotenv import load_dotenv

load_dotenv(Path(__file__).resolve().parents[1] / ".env")


class AppConfig:
    def __init__(self):
        self.jwt_secret = self._get_required("JWT_SECRET")
        self.jwt_expiration_hours = self._get_required_int("JWT_EXPIRATION_HOURS")

    def _get_required(self, key):
        value = os.getenv(key)

        if not value:
            raise ValueError(f"{key} environment variable is required.")

        return value

    def _get_required_int(self, key):
        value = self._get_required(key)

        try:
            return int(value)
        except ValueError:
            raise ValueError(f"{key} must be an integer.")

File: Projects/P2-Matt-Branch/manager-app/src/main/java/com/revature/expensemanager/config/AppConfig.java
package com.revature.expensemanager.config;

import io.github.cdimascio.dotenv.Dotenv;

public class AppConfig {
    private final String jwtSecret;
    private final int jwtExpirationHours;

    public AppConfig() {
        // A missing .env is ignored rather than fatal. Containers supply these
        // values as real environment variables, which dotenv-java falls back to
        // when a key is absent from the file, so no .env has to be shipped in
        // the image. Local runs are unaffected: their .env is still read.
        Dotenv dotenv = Dotenv.configure()
                .ignoreIfMissing()
                .load();

        this.jwtSecret = getRequired(dotenv, "JWT_SECRET");
        this.jwtExpirationHours = getRequiredInt(dotenv, "JWT_EXPIRATION_HOURS");
    }

    public String getJwtSecret() {
        return jwtSecret;
    }

    public int getJwtExpirationHours() {
        return jwtExpirationHours;
    }

    private String getRequired(Dotenv dotenv, String key) {
        String value = dotenv.get(key);

        if (value == null || value.isBlank()) {
            throw new IllegalStateException(key + " is required.");
        }

        return value;
    }

    private int getRequiredInt(Dotenv dotenv, String key) {
        String value = getRequired(dotenv, key);

        try {
            return Integer.parseInt(value);
        } catch (NumberFormatException e) {
            throw new IllegalStateException(key + " must be an integer.", e);
        }
    }
}